### ISSUE TITLE
Add Auto Versioning and Publishing to NPM Module Version

### ISSUE THREAD

**Issue by princefishthrower** [princefishthrower/react-use-please-stay#5]
Looking for someone with experience here in this.

Once automatic tests are built, I would like the following:

- Incrementally versioning each time there are code changes to the source code AND all the tests pass
- Increment only when there are changes in true source code, not other things in the repo like the README or interactive site
- How to define what type of change (major, minor, patch) and send it to such an auto versioning tool? are there tools that can determine this automatically or must they always be specified? 

Again, I have 0 experience here. All I know is that I _don't_ want to keep incrementing the version manually each time I write some code 😄

**Comment by ivanovishado** [comment]
@princefishthrower absolutely! [semantic-release](https://github.com/semantic-release/semantic-release) takes care of what you describe. I have experience using it, but not configuring it, so I guess this could be a valuable learning experience for me as well.

Let me know if I can handle this, I think it can be done in parallel with the automatic tests.

**Comment by princefishthrower** [comment]
@ivanovishado - sounds good! Go for it!

**Comment by princefishthrower** [comment]
@ivanovishado - any update here?

**Comment by ivanovishado** [comment]
@princefishthrower hey, I'll probably have some time to review this tomorrow. I'm expecting to at least open a draft PR.

**Comment by ivanovishado** [comment]
Hey @princefishthrower, sorry for the delay, I had a busy week.

I checked the `semantic-release` docs and saw that one of the requirements is to have a Continuous Integration service configured. Please let me know if you already have one since I couldn't find it in the project's files.

[Here's a list of CI services](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/ci-configuration.md#run-semantic-release-only-after-all-tests-succeeded) that may help with this decision. Let me know if you need help to configure these as well. Only after that's configured, we may continue with the `semantic-release` step.

**Comment by princefishthrower** [comment]
@ivanovishado - thanks for this. I think GitHub Actions would be the easiest, I'll add a simple one shortly, then we can get started.
